### PR TITLE
doc: update config section

### PR DESCRIPTION
**Pull request by FabianHofmann**
Update the documentation's config section. Pull out config comments to documentation.

### CODE REVIEW

**Comment by FabianHofmann** [comment]
@fneum @lisazeyen could one of you also have a look? In particular, at the documentation of the sector section at https://pypsa-eur--685.org.readthedocs.build/en/685/configuration.html#sector

File: doc/configtables/sector.csv [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+regional_co2_sequestration_potential,,,
+-- enable,--,"{true, false}",Add option for regionally-resolved geological carbon dioxide sequestration potentials based on `CO2StoP <https://setis.ec.europa.eu/european-co2-storage-database_en>`_.
+-- attribute,--,string,Name of the attribute for the sequestration potential
+-- include_onshore,--,"{true, false}",Add options for including onshore sequestration potentials
+-- min_size,Gt ,float,Any sites with lower potential than this value will be excluded
+-- max_size,Gt ,float,The maximum sequestration potential for any one site.
+-- years_of_storage,years,float,The years until potential exhausted at optimised annual rate
+co2_sequestration_potential,MtCO2/a,float,The potential of sequestering CO2 in Europe per year
+co2_sequestration_cost,currency/tCO2,float,The cost of sequestering a ton of CO2
+co2_spatial,--,"{true, false}","„Add option to spatially resolve carrier representing stored carbon dioxide. This allows for more detailed modelling of CCUTS, e.g. regarding the capturing of industrial process emissions, usage as feedstock for electrofuels, transport of carbon dioxide, and geological sequestration sites.“"
```
Comment (fneum) [inline doc/configtables/sector.csv:85]:
```suggestion
co2_spatial,--,"{true, false}","Add option to spatially resolve carrier representing stored carbon dioxide. This allows for more detailed modelling of CCUTS, e.g. regarding the capturing of industrial process emissions, usage as feedstock for electrofuels, transport of carbon dioxide, and geological sequestration sites."

```

File: doc/configtables/sector.csv [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+co2network,--,"{true, false}",Add option for planning a new carbon dioxide network
```
Comment (fneum) [inline doc/configtables/sector.csv:86]:
```suggestion
co2network,--,"{true, false}",Add option for planning a new carbon dioxide transmission network

```

File: doc/configtables/sector.csv [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+cc_fraction,--,float,The default fraction of CO2 captured with post-combustion capture
+hydrogen_underground_storage,--,"{true, false}",Add options for storing hydrogen underground. Storage potential depends regionally.
+hydrogen_underground_storage_locations,,"{onshore, nearshore, offshore}","„The location where hydrogen underground storage can be located. Onshore, nearshore, offshore means it must be located more than 50 km away from the sea, within 50 km of the sea, or within the sea itself respectively.“"
```
Comment (fneum) [inline doc/configtables/sector.csv:89]:
```suggestion
hydrogen_underground_storage_locations,,"{onshore, nearshore, offshore}","The location where hydrogen underground storage can be located. Onshore, nearshore, offshore means it must be located more than 50 km away from the sea, within 50 km of the sea, or within the sea itself respectively."

```

File: doc/configtables/sector.csv [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+ammonia,--,"{true, false, regional}","Add ammonia as a carrrier. It can be either true (copperplated NH3), false (no NH3 carrier) or ""regional"" (regionalised NH3 without network)"
+min_part_load_fischer_tropsch,per unit of p_nom ,float,The minimum unit dispatch (p_min_pu) for the Fischer-Tropsch process
+min_part_load_methanolisation,per unit of p_nom ,float,The minimum unit dispatch (p_min_pu) for the methanolisation process
```
Comment (fneum) [inline doc/configtables/sector.csv:92]:
```suggestion
min_part_load_fischer_tropsch,per unit of p_nom ,float,The minimum unit dispatch (``p_min_pu``) for the Fischer-Tropsch process
min_part_load_methanolisation,per unit of p_nom ,float,The minimum unit dispatch (``p_min_pu``) for the methanolisation process

```

File: doc/configtables/sector.csv [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+use_fischer_tropsch_waste_heat,--,"{true, false}",Add option for using waste heat of Fischer Tropsch in district heating networks
+use_fuel_cell_waste_heat,--,"{true, false}",Add option for using waste heat of fuel cells in district heating networks
+use_electrolysis_waste_heat,--,"{true, false}",Add option for using waste heat of electrolysis in district heating networks
+electricity_distribution_grid,--,"{true, false}",Add a electricity distribution grid
```
Comment (fneum) [inline doc/configtables/sector.csv:96]:
```suggestion
electricity_distribution_grid,--,"{true, false}",Add a simplified representation of the exchange capacity between transmission and distribution grid level through a link.
```

File: doc/configtables/sector.csv [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+electricity_distribution_grid_cost_factor,,,Multiplies the investment cost of the electricity distribution grid in data/costs.csv
```
Comment (fneum) [inline doc/configtables/sector.csv:97]:
```suggestion
electricity_distribution_grid_cost_factor,,,Multiplies the investment cost of the electricity distribution grid

```

File: doc/configtables/sector.csv [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+gas_network,--,"{true, false}","Add existing natural gas infrastructure, incl. LNG terminals, production and entry-points. The existing gas network is added with a lossless transport model. A length-weighted `k-edge augmentation algorithm   <https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.connectivity.edge_augmentation.k_edge_augmentation.html#networkx.algorithms.connectivity.edge_augmentation.k_edge_augmentation>`_   can be run to add new candidate gas pipelines such that all regions of the model can be connected to the gas network. When activated, all the gas demands are regionally disaggregated as well."
+H2_retrofit,--,"{true, false}","Add option for retrofiting existing pipelines to transport hydrogen. The reasoning is in accordance with the `hydrogen backbone strategy (April, 2020) p.15 <https://gasforclimate2050.eu/wp-content/uploads/2020/07/2020_European-Hydrogen-Backbone_Report.pdf>`_. 60% of original natural gas capacity could be used in cost-optimal case as H2 capacity."
+H2_retrofit_capacity_per_CH4,--,float,The ratio for H2 capacity per original CH4 capacity of retrofitted pipelines
```
Comment (fneum) [inline doc/configtables/sector.csv:102]:
```suggestion
H2_retrofit,--,"{true, false}","Add option for retrofiting existing pipelines to transport hydrogen. "
H2_retrofit_capacity_per_CH4,--,float,"The ratio for H2 capacity per original CH4 capacity of retrofitted pipelines. The `European Hydrogen Backbone (April, 2020) p.15 <https://gasforclimate2050.eu/wp-content/uploads/2020/07/2020_European-Hydrogen-Backbone_Report.pdf>`_ 60% of original natural gas capacity could be used in cost-optimal case as H2 capacity."
```

File: doc/configtables/sector.csv [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+gas_network_connectivity_upgrade ,--,float,The number of desired edge connectivity (k) in the length-weighted `k-edge augmentation algorithm <https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.connectivity.edge_augmentation.k_edge_augmentation.html#networkx.algorithms.connectivity.edge_augmentation.k_edge_augmentation>`_ used for the gas network
+gas_distribution_grid,--,"{true, false}",Add a gas distribution grid
+gas_distribution_grid_cost_factor,,,Multiplies the investment cost of the gas distribution grid in data/costs.csv
```
Comment (fneum) [inline doc/configtables/sector.csv:105]:
```suggestion
gas_distribution_grid_cost_factor,,,Multiplier for the investment cost of the gas distribution grid

```

File: doc/configtables/industry.csv [hunk 9]
```diff
@@ -0,0 +1,28 @@
+,Unit,Values,Description
+St_primary_fraction,--,Dictionary with planning horizons as keys.,The fraction of steel produced via primary route versus secondary route (scrap+EAF). Current fraction is 0.6
+DRI_fraction,--,Dictionary with planning horizons as keys.,The fraction of the primary route converted to DRI + EAF
```
Comment (fneum) [inline doc/configtables/industry.csv:3]:
```suggestion
DRI_fraction,--,Dictionary with planning horizons as keys.,The fraction of the primary route DRI + EAF

```

File: doc/configtables/industry.csv [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+MWh_elec_per_tNH3_electrolysis,--,float,"The energy amount of electricity needed to produce a ton of ammonia using Haber–Bosch process. From  `Wang et al (2018) <https://doi.org/10.1016/j.joule.2018.04.017>`_, Table 13 (air separation and HB)"
+MWh_NH3_per_MWh_H2_cracker,--,float,The energy amount of amonia needed to produce an energy amount hydrogen using ammonia cracker
+NH3_process_emissions,MtCO2/a,float,The emission of ammonia production from steam methane reforming (SMR). From UNFCCC for 2015 for EU28
+petrochemical_process_emissions,MtCO2/a,float,The emission of petrochemical production. From UNFCCC for 2015 for EU28
+HVC_primary_fraction,--,float,The fraction of today's high value chemicals (HVC) produced via primary route
+HVC_mechanical_recycling_fraction,--,float,The fraction of today's high value chemicals (HVC) produced using mechanical recycling
+HVC_chemical_recycling_fraction,--,float,The fraction of today's high value chemicals (HVC) produced using chemical recycling
```
Comment (fneum) [inline doc/configtables/industry.csv:17]:
```suggestion
HVC_primary_fraction,--,float,The fraction of high value chemicals (HVC) produced via primary route
HVC_mechanical_recycling_fraction,--,float,The fraction of high value chemicals (HVC) produced using mechanical recycling
HVC_chemical_recycling_fraction,--,float,The fraction of high value chemicals (HVC) produced using chemical recycling

```

File: doc/configtables/industry.csv [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+HVC_production_today,MtHVC/a,float,"The amount of high value chemicals (HVC) produced. This includes ethylene, propylene and BTX. From `DECHEMA (2017) <https://dechema.de/dechema_media/Downloads/Positionspapiere/Technology_study_Low_carbon_energy_and_feedstock_for_the_European_chemical_industry-p-20002750.pdf>`_, Figure 16, page 107"
+MWh_elec_per_tHVC_mechanical_recycling,MWh/tHVC,float,"The energy amount of electricity needed to produce a ton of high value chemical (HVC) using mechanical recycling. From SI of `Meys et al (2020) <https://doi.org/10.1016/j.resconrec.2020.105010>`_, Table S5, for HDPE, PP, PS, PET. LDPE would be 0.756."
+MWh_elec_per_tHVC_chemical_recycling,MWh/tHVC,float,"The energy amount of electricity needed to produce a ton of high value chemical (HVC) using chemical recycling. Value are based on pyrolysis and electric steam cracking. From `Material Economics (2019) <https://materialeconomics.com/latest-updates/industrial-transformation-2050>`_, page 125"
```
Comment (fneum) [inline doc/configtables/industry.csv:20]:
```suggestion
MWh_elec_per_tHVC_chemical_recycling,MWh/tHVC,float,"The energy amount of electricity needed to produce a ton of high value chemical (HVC) using chemical recycling. The default value is based on pyrolysis and electric steam cracking. From `Material Economics (2019) <https://materialeconomics.com/latest-updates/industrial-transformation-2050>`_, page 125"

```

**Review by fneum — CHANGES_REQUESTED** [review]
Excellent! Super helpful!

I have made small adjustments here and there.

One thing to be careful about: When the documentation says percentage, I would put e.g. 20.7 for 20.7%, but in most cases the input requires 0.207. Maybe one could clarify this.

My other comment is stylistic. Could you check if it makes sense to adjust the column widths of the tables and give more width to the description?

**Comment by FabianHofmann** [comment]
Hey @virio-andreyana, there are still some unresolved issues. Would you look into them?

File: doc/configtables/sector.csv [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+-- district_heating_loss,--,float,Percentage increase in district heat demand in urban central due to heat losses
+cluster_heat_buses,--,"{true, false}",cluster residential and service heat buses in `prepare_sector_network.py <https://github.com/PyPSA/pypsa-eur-sec/blob/master/scripts/prepare_sector_network.py>`_  to one to save memory.
+bev_dsm_restriction_value,--,float,Adding a stage of charge (SOC) limit  for battery electric vehicles (BEV) to manage its own energy demand (DSM). Located in `build_transport_demand.py <https://github.com/PyPSA/pypsa-eur-sec/blob/master/scripts/build_transport_demand.py>`_. Set to 0 for no restriction on BEV DSM
+bev_dsm_restriction_time,--,float,Time at which SOC of BEV has to be dsm_restriction_value
+transport_heating_deadband_upper,°C,float,"The minimum temperature in the vehicle. At lower temperatures, the energy required for heating in the vehicle increases."
+transport_heating_deadband_lower,°C,float,"The maximum temperature in the vehicle. At higher temperatures, the energy required for cooling in the vehicle increases."
```
[the remark this answers is not in the document]
Comment (virio-andreyana) [reply doc/configtables/sector.csv:11]:
```suggestion
transport_heating_deadband_upper,°C,float,"The maximum temperature in the vehicle. At higher temperatures, the energy required for cooling in the vehicle increases."
transport_heating_deadband_lower,°C,float,"The minimum temperature in the vehicle. At lower temperatures, the energy required for heating in the vehicle increases."

```

**Comment by virio-andreyana** [comment]
Okay, so the simple suggestion have been added. Now, only a few problem is left.

> One thing to be careful about: When the documentation says percentage, I would put e.g. 20.7 for 20.7%, but in most cases the input requires 0.207. Maybe one could clarify this.

Hmm, I believe that the reader is smart enough to know how the meaning of the percentage works. So long as they see the config default value.

> My other comment is stylistic. Could you check if it makes sense to adjust the column widths of the tables and give more width to the description?

I could try that, maybe by shortening the column for the attribute name, and values

**Comment by FabianHofmann** [comment]
thanks @virio-andreyana, perhaps we use `share` instead of `percentage`?

**Comment by FabianHofmann** [comment]
@fneum would you be fine with merging?

**Comment by fneum** [comment]
The CSVs now have quite a few empty lines. Might cause problems.

**Comment by fneum** [comment]
Otherwise, yes!

**Comment by FabianHofmann** [comment]
> The CSVs now have quite a few empty lines. Might cause problems.

@fneum thanks for looking over
@virio-andreyana this seems to be the last remaining issue

**Comment by virio-andreyana** [comment]
> The CSVs now have quite a few empty lines. Might cause problems.

Actually this is a dilemma. So what I wanted to achieve is to have all of the table to have the same spacing. But since some attribute have letters longer than 22 letters, the table shifts to the right. Making the room for the description smaller.

This is the case especially for sector.csv and industry.csv

**Comment by FabianHofmann** [comment]
> > The CSVs now have quite a few empty lines. Might cause problems.
> 
> Actually this is a dilemma. So what I wanted to achieve is to have all of the table to have the same spacing. But since some attribute have letters longer than 22 letters, the table shifts to the right. Making the room for the description smaller.
> 
> This is the case especially for sector.csv and industry.csv

Sound reasonable. Let's keep the extra lines, then. In the docs it looks good.

**Pull request merged**